### ISSUE TITLE
Sharing projects between several portia instances

### ISSUE THREAD

**Issue by cesarob** [scrapinghub/portia#493]
Hi,

I would like to run portia in serveral containers.
I have seen the projects are stored in data directory. So it looks the only way to share them is to share that directory between all the containers.

I´d like to confirm that the is no alternative storage for projects that the FS. I checked the code and it looks it is not.

Thank you,

-- César

**Comment by ruairif** [comment]
There are a few backend available for Portia. There is the default FS storage which just writes straight to the FS but there is also a git storage which is powered by [dulwich](https://github.com/jelmer/dulwich). The git storage can use a few different backends. One is the filesystem but there is also support for openstack swift and a custom mysql backend that is used to store blobs and refs.

You can find out more information about how to use these different storage options [here](http://portia.readthedocs.io/en/latest/projects.html#versioning). There is no info there on how to use swift but it should involve just using the following settings:
```python
import os

from six.moves.configparser import ConfigParser
from six.moves import StringIO

SWIFT_ROOT = '/'
parser = ConfigParser()
cfg = StringIO("""
# Configuration file sample
[swift]
# Authentication URL (Keystone or Swift)
auth_url = http://127.0.0.1:5000/v2.0
# Authentication version to use
auth_ver = 2
# The tenant and username separated by a semicolon
username = admin;admin
# The user password
password = pass
# The Object storage region to use (auth v2) (Default RegionOne)
region_name = RegionOne
# The Object storage endpoint URL to use (auth v2) (Default internalURL)
endpoint_type = internalURL
# Concurrency to use for parallel tasks (Default 10)
concurrency = 10
# Size of the HTTP pool (Default 10)
http_pool_length = 10
# Timeout delay for HTTP connections (Default 20)
http_timeout = 20
# Chunk size to read from pack (Bytes) (Default 12228)
chunk_length = 12228
# Cache size (MBytes) (Default 20)
cache_length = 20
""")
parser.readfp(cfg)

SPEC_FACTORY = {
    'PROJECT_SPEC': 'slyd.gitstorage.projectspec.ProjectSpec',
    'PROJECT_MANAGER': 'slyd.gitstorage.projects.ProjectsManager',
    'PARAMS': {
        'storage_backend': 'dulwich.fsrepo.FsRepo',
        'root': os.environ.get('PORTIA_SWIFT_ROOT', SWIFT_ROOT),
        'conf': parser
    },
    'CAPABILITIES': {
        'version_control': True,
        'create_projects': True,
        'delete_projects': True,
        'rename_projects': True
    }
}
```

If you have any problems please post them and hopefully we can get this working for you.

**Comment by cesarob** [comment]
A solution could be to use NFS.

We decided not to use it as it would complicate our infraestructure... So we have implemented an alternative approach using S3. When a instance starts it syncs with the S3 bucket (we didn't want to use s3fs as we didn't have good experiences in the past).
We update the S3 butcket using inotify.

For those using AWS like us just tell you that they just opened [EFS](https://aws.amazon.com/es/efs/).

I think this issue can be closed. As it was really a question, and possible solutions are offered.
